packages/contracts/contracts/minter-suite/Minters/MinterSetPriceV5Merkle.sol

```diff
@@ -0,0 +1,650 @@
+
+    /// minter version for this minter
+    string public constant minterVersion = "v5.0.0";
+
+    uint256 constant ONE_MILLION = 1_000_000;
+
+    /// contractAddress => projectId => base project config
+    mapping(address => mapping(uint256 => ProjectConfig)) public projectConfig;
+
+    /// contractAddress => projectId => purchaser address => qty of mints purchased for project
+    mapping(address => mapping(uint256 => mapping(address => uint256)))
+        public projectUserMintInvocations;
+
+    // STATE VARIABLES FOR SplitFundsLib
+    // contractAddress => IsEngineCache
+    mapping(address => SplitFundsLib.IsEngineCache) private _isEngineCaches;
+
+    // STATE VARIABLES FOR MerkleLib
+    /// Delegation registry address
+    address public immutable delegationRegistryAddress;
+    /// Delegation registry address
+    IDelegationRegistry private immutable delegationRegistryContract;
+
+    /// contractAddress => projectId => merkle specific project config
+    mapping(address => mapping(uint256 => MerkleLib.MerkleProjectConfig))
+        public merkleProjectConfig;
+
+    // STATE VARIABLES FOR MaxInvocationsLib
+    /// contractAddress => projectId => max invocations specific project config
+    mapping(address => mapping(uint256 => MaxInvocationsLib.MaxInvocationsProjectConfig))
+        public maxInvocationsProjectConfig;
+
+    function _onlyArtist(
+        uint256 _projectId,
+        address _coreContract
+    ) internal view {
+        require(
+            msg.sender ==
+                IGenArt721CoreContractV3_Base(_coreContract)
+                    .projectIdToArtistAddress(_projectId),
+            "Only Artist"
+        );
+    }
+
+    /**
+     * @notice Initializes contract to be a Filtered Minter for
+     * `_minterFilter`, integrated with Art Blocks core contract
+     * at address `_genArt721Address`.
+     * @param _minterFilter Minter filter for which this will be a
+     * filtered minter.
+     */
+    constructor(
+        address _minterFilter,
+        address _delegationRegistryAddress
+    ) ReentrancyGuard() {
+        minterFilterAddress = _minterFilter;
+        minterFilter = IMinterFilterV1(_minterFilter);
+
+        delegationRegistryAddress = _delegationRegistryAddress;
+        emit DelegationRegistryUpdated(_delegationRegistryAddress);
+        delegationRegistryContract = IDelegationRegistry(
+            _delegationRegistryAddress
+        );
+        // broadcast default max invocations per address for this minter
+        emit DefaultMaxInvocationsPerAddress(
+            MerkleLib.DEFAULT_MAX_INVOCATIONS_PER_ADDRESS
+        );
+    }
+
+    /**
+     * @notice Returns whether or not the provided address `_coreContract`
+     * is an Art Blocks Engine core contract. Caches the result for future access.
+     * @param _coreContract Address of the core contract to check.
+     */
+    function _isEngine(address _coreContract) internal returns (bool) {
```

Comment:
Should this be used at all vs. just using the library?

Reply:
moved this entirely to the lib, except for a new view func, `isEngineView`, on the minter to allow for easy viewing of this data (non-modifying vs the funcs that set cache)